library/spreadsheet.inc.php

```diff
@@ -124,31 +124,46 @@ function real2form($fldval)
 
         // If updating an existing form...
         if ($formid) {
-            sqlStatement("UPDATE form_$spreadsheet_form_name SET "      .
-            "value = '$form_completed|$start_date|$template_name' " .
-            "WHERE id = '$formid' AND rownbr = -1 AND colnbr = -1");
-            sqlStatement("DELETE FROM form_$spreadsheet_form_name WHERE " .
-              "id = '$formid' AND rownbr >= 0 AND colnbr >= 0");
+            sqlStatement(
+                "UPDATE " . escape_table_name('form_' . $spreadsheet_form_name) .
+                " SET value = ? WHERE id = ? AND rownbr = -1 AND colnbr = -1",
+                array(
+                    $form_completed|$start_date|$template_name,
```

Comment:
Change to:
```
$form_completed . '|' . $start_date . '|' . $template_name
```

Reply:
I thought it was a bitwise operation initially. This makes more sense though. Thanks.